src/Time/Units.hs

```diff
@@ -132,6 +135,27 @@ type AllTimes =
    , Millisecond , Microsecond, Nanosecond, Picosecond
    ]
 
+{- | Creates the list of time units in descending order by provided
+the highest and the lowest bound of the desired list.
+
+__Usage example:__
+
+>>> seriesF @(Hour ... Second) $ hour 3 +: minute 5 +:sec 3 +: ms 123
+"3h5m3s"
+
+-}
+type family (...) (a :: Rat) (b :: Rat) where
+    x ... y = TakeWhileNot y (DropWhileNot x AllTimes)
+
+-- Drops wile not the required time unit in 'AllTimes'.
+type family DropWhileNot (a :: Rat) (c :: [Rat]) where
+    DropWhileNot x '[] = '[]
+    DropWhileNot x (u ': units) = If (u ==% x) (u ': units) (DropWhileNot x units)
+
+-- Takes while not equal to the provided bound.
+type family TakeWhileNot (a :: Rat) (c :: [Rat]) where
+    TakeWhileNot x '[] = '[]
+    TakeWhileNot x (u ': units)  = If (u ==% x) '[u] (u ': TakeWhileNot x units)
```

Comment:
Looks cool :+1: I like this solution! I had in mind something much less elegant... So I really like the current one :slightly_smiling_face: Though here is redundant space before `=`.